pymc/tests/test_aesaraf.py

```diff
@@ -704,3 +706,126 @@ def update(self, node):
         fn = compile_pymc(inputs=[], outputs=dummy_x)
         assert fn() == 2.0
         assert fn() == 3.0
+
+
+class TestCompileForwardSampler:
```

Comment:
We need tests with variables created by `SymbolicDistributions` like `pm.Censored` and `pm.Mixture`, which do not return pure `RandomVariables` but should be treated as such for the sake of forward sampling.

Reply:
What should we test for here? Have a latent mixture that is volatile or not?

Reply:
I don't know what the test should be, but my concern is that Mixture takes as inputs random variables that are not in the trace by definition, but we don't want to resample Mixtures if they are in the trace, right?

Reply:
This indeed is hard to fix. Is there a way for me to test if the RV was created by a symbolic distribution? Maybe I’ll have to add a new type like SymbolicRV and test against that

Reply:
Those RVs shouldn't be in model.basic_RVs so maybe you can use that

Reply:
I didn't want to go check the `model` instance. I wanted to make the aesara variables have all the information necessary to make this work on their own, without having to resort to external book keeping. To make this happen, I had to go quite meta-python. I had to write a metaclass that wraps the `SymbolicDistribution.rv_op` classmethod. This was really tricky because classmethods are descriptors, so they can't be wrapped as easily as regular functions or methods. The good news is that I managed to figure out how to wrap the descriptor and add a simple boolean tag to the output variable of `rv_op`, so the code now checks if the clients are pure `RandomVariable`s or if they were created using a `SymbolicDistribution.rv_op`. We could actually use this new tag to fix the behavior point out in #5765.

Reply:
This bookkeeping is precisely the one and only job of `Model`. My issue with adding tag information is that it makes graph manipulation using standard Aesara features more dangerous because we have to make sure to duplicate all the tag information that is important in the context of PyMC in all rewrites, so that other routines will work properly with the optimized graphs.